keps/sig-node/3695-pod-resources-for-dra/README.md

```diff
@@ -0,0 +1,372 @@
+    repeated ContainerMemory memory = 4;
+    repeated CDIDevices cdi_devices = 5;
+}
+
+// ContainerMemory contains information about memory and hugepages assigned to a container
+message ContainerMemory {
+    string memory_type = 1;
+    uint64 size = 2;
+    TopologyInfo topology = 3;
+}
+
+// ContainerDevices contains information about the devices assigned to a container by device plugin
+message ContainerDevices {
+    string resource_name = 1;
+    repeated string device_ids = 2;
+    TopologyInfo topology = 3;
+}
+
+// Topology describes hardware topology of the resource
+message TopologyInfo {
+    repeated NUMANode nodes = 1;
+}
+
+// NUMA representation of NUMA node
+message NUMANode {
+    int64 ID = 1;
+}
+
+// CDIDevices contains information about the devices assigned to a container by DRA
+message CDIDevices {
+    string driver_name = 1;
+    string resource_class = 2;
+    repeated string device_ids = 1;
+}
+
+// GetPodResourcesRequest contains information about the pod
+message GetPodResourcesRequest {
+    string name = 1;
+    string namespace = 2;
+}
+
+// GetPodResourcesResponse contains information about the pod the devices
+message GetPodResourcesResponse {
+    PodResources pod_resources = 1;
+}
+
+```
+
+
+##### Unit tests
+
+- `k8s.io/kubernetes/pkg/kubelet/apis/podresources`: `01-24-2023` - `61.5%`
+
+##### Integration tests
+
+ These cases will be added in the existing integration tests:
+  - Feature gate enable/disable tests.
+  - Get API work with DRA and device plugin.
+  - List API work with DRA and Device plugin.
+
+##### e2e tests
+
+These cases will be added in the existing e2e tests:
+  - Feature gate enable/disable tests.
+  - Get API work with DRA and device plugin.
+  - List API work with DRA and Device plugin.
+
+### Graduation Criteria
+
+#### Alpha
+
+- [X] Feature implemented behind a feature flag.
+- [X] e2e tests completed and enabled.
+
+#### Beta
+
+- [ ] Gather feedback from consumers of the DRA feature and k8snetworkplumbingwg working group
+- [ ] No major bugs reported in the previous cycle.
+
+#### GA
+
+- [ ] Allowing time for feedback (1 year).
+- [ ] Risks have been addressed.
+
+### Upgrade / Downgrade Strategy
+
+With gRPC the version is part of the service name.
+Old versions and new versions should always be served and listened by the kubelet.
+
+To a cluster admin upgrading to the newest API version, means upgrading Kubernetes to a newer version as well as upgrading the monitoring component.
+
+To a vendor changes in the API should always be backwards compatible.
+
+### Version Skew Strategy
+
+Kubelet will always be backwards compatible, so going forward existing plugins are not expected to break.
+
+
+## Production Readiness Review Questionnaire
+### Feature Enablement and Rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+- [x] Feature gate (also fill in values in `kep.yaml`)
+  - Feature gate name: PodResourcesCDIDevices.
```

Comment:
It could be surprising that the `Get` API availability is governed by a feature gate whose name doesn't hint back to `Get` in any way. This is probably a byproduct of the fact this KEP bundles together two enhancements both useful for CNI metaplugins but pretty indpendent with each other. I'd ask for directions to PRReviewers or approvers, as I can think of any good solution here.

Reply:
Good point, I think it would probably be more apt to have another feature gate to guard the enablement/disablement of the `Get` endpoint. Agree that PRR reviewers should be able to provide input here.

Reply:
Agreed, we should have 2 separate feature gates here.